[inductor] Fix flaky tests in test_pad_mm

aakhundov opened the pull request:
Summary: When run internally, some tests in `test_pad_mm.py` requiring big enough GPU to run `max_autotune=True` fail, as they're getting a smaller GPU than they need. Here we add `skipTest`s to skip the tests in these (rare) circumstances.

Differential Revision: D60192586


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

eellison commented on test/inductor/test_pad_mm.py:
```diff
@@ -11,14 +11,17 @@
 
 
 class PadMMTest(TestCase):
     @inductor_config.patch(max_autotune=True, max_autotune_gemm_backends="TRITON")
```
Can you do this in the testSetup instead of in each test ? or just do it as a decorator on the class itself?

aakhundov replied:
@eellison neither option works directly. Decorators are executed on GPU-less host in RE so `is_big_gpu` just fails (I think). `testSetup` would affect all tests, but some tests here seem to not require `max_autotune=True` / TRITON backend to run. I could split the tests into two classes, but not sure what the behavior of `run_tests` would be: will it run tests from all classes in the file?

aakhundov replied:
On a second thought, I think, it's fine to check the `is_big_gpu` in `setUp`. Let me update the PR.

aakhundov committed:
[inductor] Fix flaky tests in test_pad_mm (#131699)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/131699

When run internally, some tests in `test_pad_mm.py` requiring big enough GPU to run `max_autotune=True` fail, as they're getting a smaller GPU than they need. Here we add `skipTest`s to skip the tests in these (rare) circumstances.

Test Plan:
```
buck2 test -j 50 'fbcode//mode/opt' fbcode//caffe2/test/inductor:pad_mm -- --exact 'caffe2/test/inductor:pad_mm - test_pad_addmm_dyn_m (caffe2.test.inductor.test_pad_mm.PadMMTest)' --run-disabled --stress-runs 50 --record-results
```

https://www.internalfb.com/intern/testinfra/testrun/15481123761759967

Reviewed By: eellison, shunting314, chenyang78

Differential Revision: D60192586
